client/src/pages/OfferForm/index.tsx

```diff
@@ -0,0 +1,106 @@
+import { useState } from 'react'
+import './OfferForm.css'
+import { OfferData } from '../../types/offerTypes'
+
+interface OfferFormProps {
+  task: {
+    id: string
+    title: string
+  }
+  onSubmit: (offerData: OfferData) => void
+  onCancel: () => void
+}
+
+const OfferForm = ({ task, onSubmit, onCancel }: OfferFormProps) => {
+  const [offerData, setOfferData] = useState<OfferData>({
+    offer_description: '',
+    offer_media: null,
+    posting_time: '',
+    price: '',
+    offer_status: 'pending',
+  })
+
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => {
+    setOfferData({
+      ...offerData,
+      [event.target.name]: event.target.value,
+    })
+  }
+
+  const handleFileChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const file = event.target.files?.[0] || null
+    setOfferData({
+      ...offerData,
+      offer_media: file,
+    })
+  }
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault()
+    const offerPayload = {
+      job_id: task.id,
+      worker_id: '1',
+      offer_description: offerData.offer_description,
+      posting_time: offerData.posting_time,
+      price: offerData.price,
+      offer_status: offerData.offer_status,
+      offer_media: offerData.offer_media,
+    }
+    onSubmit(offerPayload)
+  }
+
+  return (
+    <div className='offer-form-content'>
+      <h2>Make an Offer for {task.title}</h2>
+      <form onSubmit={handleSubmit}>
+        <textarea
+          name='offer_description'
+          placeholder='Offer Description'
+          value={offerData.offer_description}
+          onChange={handleChange}
+          required
+        />
+        <input
+          type='file'
+          name='offer_media'
+          accept='.jpg, .jpeg, .png, .gif, .pdf, .zip, .rar'
+          multiple
+          onChange={handleFileChange}
+          required
+        />
```

Comment:
You can give a label to each input for better customer UI and DOM.

```javascript
        <label htmlFor="offer_media">Upload Media</label>
        <input
          type="file"
          id="offer_media"
          name="offer_media"
          accept=".jpg, .jpeg, .png, .gif, .pdf, .zip, .rar"
          multiple
          onChange={handleFileChange}
          required
        />
```

Reply:
Same with the others.

Reply:
I've added the labels and put hidden 'posting_time', don't think that it should be the users handling that

Reply:
![Screenshot 2024-07-15 at 18 35 19](https://github.com/user-attachments/assets/02674ff7-e744-428e-a4bc-27505a0c4614)